ntp-daemon/src/system.rs

```diff
@@ -0,0 +1,184 @@
+use crate::peer::{MsgForSystem, PeerChannels, PeerIndex, PeerTask, ResetEpoch};
+use ntp_os_clock::UnixNtpClock;
+use ntp_proto::{
+    ClockController, ClockUpdateResult, FilterAndCombine, NtpInstant, PeerSnapshot, SystemConfig,
+    SystemSnapshot,
+};
+use tracing::info;
+
+use std::{error::Error, sync::Arc};
+use tokio::sync::{mpsc, watch};
+
+/// Spawn the NTP daemon
+pub async fn spawn(config: &SystemConfig, peer_addresses: &[&str]) -> Result<(), Box<dyn Error>> {
+    // shares the system state with all peers
+    let global_system_snapshot = Arc::new(tokio::sync::RwLock::new(SystemSnapshot::default()));
+
+    // send the reset signal to all peers
+    let reset_epoch: ResetEpoch = ResetEpoch::default();
+    let (reset_tx, reset_rx) = watch::channel::<ResetEpoch>(reset_epoch);
+
+    // receive peer snapshots from all peers
+    let (msg_for_system_tx, msg_for_system_rx) = mpsc::channel::<MsgForSystem>(32);
+
+    for (index, address) in peer_addresses.iter().enumerate() {
+        let channels = PeerChannels {
+            msg_for_system_sender: msg_for_system_tx.clone(),
+            system_snapshots: global_system_snapshot.clone(),
+            reset: reset_rx.clone(),
+        };
+
+        PeerTask::spawn(
+            PeerIndex { index },
+            address,
+            UnixNtpClock::new(),
+            *config,
+            channels,
+        )
+        .await?;
+    }
+
+    let mut peers = Peers::new(peer_addresses.len());
+
+    run(
+        config,
+        &mut peers,
+        reset_epoch,
+        global_system_snapshot,
+        msg_for_system_rx,
+        reset_tx,
```

Comment:
Minor, however if we want to change it now is the time, but I think it might be nicer to not pass these as parameters to run, but instead have a daemon struct that contains most of these, with run a method on it.

Reply:
Ok, then we will leave it as is.